MetaMask/metamask-mobile
# feat: decouple account selector from qr code connector

## **Description**

In order to enable multiple account select in ledger hardware wallet, 
This PR is focus on decoupling the `Account selector component` from QR hardware wallet for reusable purpose

The changes should not impact to any UI / operations differences compare to existing QR hardware wallet account selector

For clean code purpose, the changes also decoupling the method `toBlockExplorer` from account selector component into a new hook `useBlockExplorer`

**Logical flow**
![image](https://github.com/MetaMask/metamask-mobile/assets/102275989/ef5715e1-2603-49f0-96e8-6c2c394fb9a5)

## **Related issues**

issue : [109](https://app.zenhub.com/workspaces/cet-metamask-hardware-wallets-6566ee6aedc46007d5a260bb/issues/zh/109)

## **Manual testing steps**

1. Go to main screen
2. Tab current account to open account drawer
3. Click add account or hardware account
4. Choose add hardware account
5. Choose QR based
6. Scan Qr wallet
7. Account selector should appear after scan success
8. Account selector should able to perform select/un select account, unlock accounts, next page, prev pages operation as what it has to be in previous version

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
